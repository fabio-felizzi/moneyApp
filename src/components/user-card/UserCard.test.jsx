import React from 'react';
import { shallow } from 'enzyme';
import UserCard from './UserCard';

const componentProps = {
    className: 'user-card',
    name: 'test-card',
    dob: '12/12/12',
    employmentStatus: 'full-time',
    income: 50000,
    houseNumber: 123,
    postCode: 'EC1 1AB',
    submitUser: () => {},
    id: 1,
};

const componentShallow = shallow(<UserCard {...componentProps} />);

describe('UserCard component', () => {
    describe('Component Shapshot', () => {
        test('should match stored snapshot', () => {
            expect(componentShallow).toMatchSnapshot();
        });
    });

    describe('Component Render', () => {
        test('should render the title', () => {
            expect(componentShallow.find('h3')).toHaveLength(1);
            expect(componentShallow.find('h3').text()).toBe(componentProps.name);
        });

        test('should render a ul', () => {
            expect(componentShallow.find('ul')).toHaveLength(1);
        });

        test('should render an li with the dob prop', () => {
            expect(componentShallow.find('#dob').text()).toBe(`D.O.B: ${componentProps.dob}`);
        });

        test('should render an li with the employmentStaus prop', () => {
            expect(componentShallow.find('#status').text()).toBe(`Employment Status: ${componentProps.employmentStatus}`);
        });

        test('should render an li with the income prop', () => {
            expect(componentShallow.find('#income').text()).toBe(`Income: ${componentProps.income}`);
        });

        test('should render an li with the houseNumber prop', () => {
            expect(componentShallow.find('#houseNo').text()).toBe(`House No: ${componentProps.houseNumber}`);
        });

        test('should render an li with the postCode prop', () => {
            expect(componentShallow.find('#postCode').text()).toBe(`Postcode: ${componentProps.postCode}`);
        });
    });
});
