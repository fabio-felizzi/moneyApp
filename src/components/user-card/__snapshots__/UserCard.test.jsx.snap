// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Component Shapshot should match stored snapshot 1`] = `
<div
  className="user user-card"
>
  <h3>
    test-card
  </h3>
  <ul>
    <li
      id="dob"
    >
      D.O.B: 12/12/12
    </li>
    <li
      id="status"
    >
      Employment Status: full-time
    </li>
    <li
      id="income"
    >
      Income: 50000
    </li>
    <li
      id="houseNo"
    >
      House No: 123
    </li>
    <li
      id="postCode"
    >
      Postcode: EC1 1AB
    </li>
  </ul>
  <button
    onClick={[Function]}
    type="button"
  >
    Search for cards
  </button>
</div>
`;
